diff --git a/lib/lowlevel.c b/lib/lowlevel.c
index 02060a8..3b142bf 100644
--- a/lib/lowlevel.c
+++ b/lib/lowlevel.c
@@ -246,7 +246,7 @@ int ll_open(struct afp_volume * volume,
     if (access_mode == O_RDONLY) {
         aflags |= AFP_OPENFORK_ALLOWREAD;
     } else if (access_mode == O_WRONLY) {
-        aflags |= AFP_OPENFORK_ALLOWWRITE;
+        aflags |= (AFP_OPENFORK_ALLOWREAD | AFP_OPENFORK_ALLOWWRITE);
     } else if (access_mode == O_RDWR) {
         aflags |= (AFP_OPENFORK_ALLOWREAD | AFP_OPENFORK_ALLOWWRITE);
     }
